File: migratory-birds.java
class Result {
    public static int migratoryBirds(List<Integer> arr) {
        HashMap<Integer,Integer> ar = new HashMap<>();
        for(Integer num : arr){
            if(ar.containsKey(num)){
                int val = ar.getOrDefault(num,0);
                ar.put(num, val+1);
            }
            else{
                ar.put(num, 1);
            }
        }
        int mx=0;
        int ans=0;
        for(Integer key:ar.keySet()){
            int value=ar.get(key);
            if(value>mx){
                mx=value;
                ans=key;
            }
        }
        return ans;
    }
}

File: migratory-birds.py
from collections import defaultdict

def migratoryBirds(arr):
    hash_map = defaultdict(int)
    for num in arr:
        hash_map[num] +=1
        
    max_count = 0
    ans = float('inf')
        
    for bird_id, count in hash_map.items():
        if count > max_count:
            max_count = count
            ans = bird_id
        elif count == max_count:
            if bird_id < ans:
                ans = bird_id
    return ans